src/core/dbus-unit.c

```diff
@@ -1880,8 +1880,10 @@ int bus_unit_queue_job(
         /* dbus-broker issues StartUnit for activation requests, so let's apply the same check
          * used in signal_activation_request(). */
         if (type == JOB_START && u->type == UNIT_SERVICE &&
-            SERVICE(u)->type == SERVICE_DBUS && !manager_dbus_is_running(u->manager))
-                return sd_bus_error_set(error, BUS_ERROR_SHUTTING_DOWN, "Refusing activation, D-Bus is not running.");
+            SERVICE(u)->type == SERVICE_DBUS &&
+            (manager_unit_inactive_or_pending(u->manager, SPECIAL_DBUS_SERVICE) ||
```

Comment:
Apparently in my PR I overlooked the fact that dbus can be manually stopped...was thinking it shouldn't be. But I think this change still suffers from the same problem, i.e. if dbus is inactive then we'll refuse the start?

Rethinking about this, what we want to avoid is just job type conflict here. So maybe `unit_stop_pending(manager_get_unit..., DBUS...)` is sufficient?

Reply:
Oh, a recent issue (https://github.com/systemd/systemd/issues/27549) shows that we can't handle the case when a service is already stopping but we try to start it through dependency. Then we can check both the service state and pending jobs here until that gets solved.

Reply:
> But I think this change still suffers from the same problem, i.e. if dbus is inactive then we'll refuse the start?

I think UNIT_DEACTIVATING state should also be checked.

Reply:
> I think UNIT_DEACTIVATING state should also be checked.

There's no need tho? When the stop job is already running, a new start job would just wait until that completes. As it doesn't cause job type conflict, it won't be canceled.

Reply:
@Geass-LL Could you please test whether a running stop job also interrupts the reboot?

Reply:
> @Geass-LL Could you please test whether a running stop job also interrupts the reboot?

If there is a running stop job, systemd will wait for this job until timeout, but this job won't interrupt the reboot.

Reply:
> > I think UNIT_DEACTIVATING state should also be checked.
> 
> There's no need tho? When the stop job is already running, a new start job would just wait until that completes. As it doesn't cause job type conflict, it won't be canceled.

Guess you're right. If a service goes to UNIT_DEACTIVATING state but doesn't have a pending stop job, this means this service is not stopped by systemd, i.e. killed by SIGTERM sent from other process. Here, we only want to refuse the service activation request when dbus-broker is being stopped by systemd's job.